src/clr-angular/data/datagrid/interfaces/column-state.interface.ts

```diff
@@ -0,0 +1,12 @@
+/*
+ * Copyright (c) 2016-2018 VMware, Inc. All Rights Reserved.
+ * This software is released under MIT license.
+ * The full license information can be found in LICENSE in the root directory of this project.
+ */
+import { DatagridColumnChanges } from '../enums/column-changes.enum';
+
+export interface DatagridColumnState {
+  actions?: DatagridColumnChanges[];
```

Comment:
This looks like it saves the specific states to replay them on demand. Two comments. 

1. They way Header/Cell renderers react to changes is switching on each action. Is there any concern about that growing to verbose and or complicated requireing many helper utility methods in both renderers? 
2. naming nit: is actions the best way to descibe these changes that need to be applied?

Reply:
Would you prefer `changes` instead of `actions`?

Reply:
`changes` makes more sense to me since its an array of `DatagridColumnChanges`'s. It wouldn't be something that I would feel strong enough about to block if actions is more natural to everyone else.